### PR TITLE
Add ISO8601 parsing support to qx.lang.normalize.Date #7175

### PR DESCRIPTION
**Pull request by schnepe2**
This is the PR in response to the discussion at https://github.com/qooxdoo/qooxdoo/pull/9112

Note: That this PR does only provide one (standard confirm) `parse`. (see discussion above)

Note: The check `qx.bom.client.EcmaScript#getDateParse` might return `false` when run in a timezone equal to GMT, but then the only thing that happens is that the (already compliant) `Date.parse` method might get replaced by our (also compliant) `Date.parse`. So it's fail-save.
But if someone has a better idea,...bring it on :grinning:

### CODE REVIEW

File: framework/source/class/qx/lang/normalize/Date.js [hunk 1]
```diff
@@ -37,13 +37,80 @@ qx.Bootstrap.define("qx.lang.normalize.Date", {
...
+
+      // Return the number of milliseconds since Epoch
+      return Date.UTC( captureGroups[1], captureGroups[2], captureGroups[3],
+                       captureGroups[4], captureGroups[5] + minutesOffset,
+                       captureGroups[6], captureGroups[7] );
     }
   },
 
-  defer : function(statics) {
+  defer : function (statics)
```
Comment (oetiker) [inline framework/source/class/qx/lang/normalize/Date.js]: I get that you prefer a space after 'function' but that does not seem to be the style in this file in general

Comment (schnepe2) [inline framework/source/class/qx/lang/normalize/Date.js]: Yeah, that's what I like the best: declaration/definition **with** space, calls **without** space :+1: 
But usually I only apply this to methods that I've changed, not on a whole module. It confuses people when looking at the history (issue-related vs. other changes).

**Comment by oetiker** [comment]
can you sqash please ? then I think we are ready for a vote

**Comment by schnepe2** [comment]
@oetiker sqash?! ...sorry "Je ne comprends pas" / "No entiendo" / "Non capisco" / "Eu não entendo"...

**Comment by oetiker** [comment]
```
git rebase -i HEAD~2
```

then edit the list so that only one commit remains

```
git push --force
```

see also https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

the reason for that is that cherry picking changes patches from one branch to another is much simpler like that and unless the individual commits are important for understanding the PR, then it makes sense to suqash them.

**Comment by schnepe2** [comment]
O.K. done that, although I think it made things worse.
git is a pain in the a** for beginners! I hate it for being such unnecessary complicated!

**Comment by oetiker** [comment]
hmmm that definitely went into the wrong direction ... now you have 4 commits and half of johns latest patch in your branch ...

yes git takes some getting used to :) but once you do it works great

**Comment by schnepe2** [comment]
So...what are you gonna do?

**Comment by oetiker** [comment]
@schnepe2 I can not do anything about it ... since this a PR from your branch ... but you can try again :)
OR just create a new PR ... but getting the whole squashing to work is pretty rewarding as it allows you to 'fix' little mistakes without cluttering the changelog which I find marvelous ... I rarely get it right in the first attempt.

**Comment by woprandi** [comment]
@schnepe2 `git rebase -i HEAD~4 `, in the 3 last lines, replace "pick" by "f" to merge them to the first commit without editing commit message
Then with `git log -p` if you have evrything in the last commit
Then `git push -f`

**Comment by oetiker** [comment]
note though the two commits from @johnspackman that you somehow acquired into the PR should not be there nor should they be squashed ... it seems to me a problem with the branch point you may have to bring this forward to the tip of master `git pull upstream master` (I think)

**Comment by woprandi** [comment]
Just a `git rebase origin/master` ?

**Comment by schnepe2** [comment]
...**that's** what I am talking about :grinning: ... 3 comments for one action and no one is sure what will happen on my side :laughing: 
I think I'll rebase my fork, create a complete new branch for this and do another PR...phew

**Comment by zaucker** [comment]
Shouldn't this PR be closed as it is replaced by PR9145?

**Comment by level420** [comment]
@zaucker @schnepe2: yes it should be closed, optimally @schnepe2 should do this.
But if you like...

**Comment by schnepe2** [comment]
Closed this PR as it **is** replaced by PR #9145

**Pull request closed without merging**